Dashboard: replace Base64 with atob/btoa

mistydemeo opened the pull request:
This replaces the vendored Base64 code we're using with thin wrappers on top of the builtin `atob` and `btoa` functions that are in Firefox and Chrome, as well as IE 10 and newer. The wrappers are capable of safely roundtripping both Unicode strings and mojibake.

This resolves the issues with the Base64 code we were using misdecoding certain regions of Unicode characters.

mistydemeo commented:
Hm, I suppose I could reduce the size of the diff by simply naming the new functions `Base64.encode` and `Base64.decode`, now that I think about it.

Hwesta commented:
:+1:  either way.

Keeping Base64.[en|de]code is a smaller diff, base64_to_utf8 is more clear.

Commit:
Dashboard: replace Base64 with atob/btoa

The current JavaScript-based base64-encoding library we're using in
Archivematica and the storage service mishandles certain Unicode
codepoints. This corrupts certain strings when decoding/encoding, and
makes it impossible to roundtrip all data. For example:

	Base64.decode("8J+Yhy3wn5iP") // => "ߘ筰", not "😇-😏"

This replaces the vendored Base64 library with thin wrappers around the
native JavaScript atob/btoa functions. Those functions are available in
Firefox, Chrome, Safari, and Internet Explorer 10+.

Fixes #7698.

mistydemeo commented:
Updated to minimize the diff.

Hwesta commented:
Still :+1:
